server/adaptors/integrations/types.ts

```diff
@@ -27,29 +17,83 @@ interface IntegrationConfig {
   author?: string;
   description?: string;
   sourceUrl?: string;
-  statics?: {
-    logo?: StaticAsset;
-    gallery?: StaticAsset[];
-    darkModeLogo?: StaticAsset;
-    darkModeGallery?: StaticAsset[];
-  };
+  statics?: IntegrationStatics;
   components: IntegrationComponent[];
   assets: IntegrationAssets;
   sampleData?: {
     path: string;
   };
 }
 
+// IntegrationConfig extended with local copies of all data
+interface SerializedIntegration extends IntegrationConfig {
+  statics?: SerializedIntegrationStatics;
+  components: SerializedIntegrationComponent[];
+  assets: SerializedIntegrationAssets;
+  sampleData?: {
+    path: string;
+    data: string;
+  };
+}
+
+interface IntegrationStatics {
+  logo?: StaticAsset;
+  gallery?: StaticAsset[];
+  darkModeLogo?: StaticAsset;
+  darkModeGallery?: StaticAsset[];
+}
+
+interface SerializedIntegrationStatics {
+  logo?: SerializedStaticAsset;
+  gallery?: SerializedStaticAsset[];
+  darkModeLogo?: SerializedStaticAsset;
+  darkModeGallery?: SerializedStaticAsset[];
+}
+
+interface IntegrationAssets {
+  savedObjects?: {
+    name: string;
+    version: string;
+  };
+  queries?: Array<{
+    name: string;
+    version: string;
+    language: string;
+  }>;
+}
+
+interface SerializedIntegrationAssets extends IntegrationAssets {
+  savedObjects?: {
+    name: string;
+    version: string;
+    data: string;
```

Comment:
Saving them as strings allows for more uniformity in parsing logic and has less validation complexity. We can fall back to OSD core's validation for parsing saved objects since I've switched to their parser, and the schema format would also be tricky to define (we generally just parse it and send it to the backend -- maybe the validation logic for that is exported too?). It might be more clear and play more nicely with autoformatting, but I'm not sure if it's worth the added effort given our use case. 🤔